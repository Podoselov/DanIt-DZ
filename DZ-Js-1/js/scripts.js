'use strict';
let userName = '';
let userAge = '';
do {
  userName = prompt('Enter your name', `${userName}`);
  userAge = prompt('How old are you', `${userAge}`);
} while (valid(userName, userAge));

function valid(name, age) {
  return (
    name === null ||
    name.trim() === '' ||
    age === null ||
    typeof +age !== 'number' ||
    Number.isNaN(+age) ||
    age.trim() === ''
  );
}

function authorization(name, age) {
  let errorMessage = 'You are not allowed to visit this website';
  let allowed = `Welcome,  ${name} !`;
  if (age < 18) {
    alert(`${errorMessage}`);
  } else if (age > 22) {
    alert(`${allowed}`);
  } else {
    let continuationQuestion = confirm('Are you sure you want to continue?')
      ? alert(`${allowed}`)
      : alert(`${errorMessage}`);
  }
}

authorization(userName, userAge);
