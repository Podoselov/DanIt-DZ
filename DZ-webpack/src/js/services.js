$(function showActiveEl() {
  $('.services__link').on('click', function (e) {
    e.preventDefault();
    $('.services__link').each(function (el) {
      $(this).removeClass('active-el');
    });
    $(this).addClass('active-el');
    let dataValue = $(this).data('name');
    $('.services__description').each(function (el) {
      $(this).removeClass('services__description--show');
      if ($(this).data('name') === dataValue) {
        $(this).addClass('services__description--show');
      }
    });
  });
});
