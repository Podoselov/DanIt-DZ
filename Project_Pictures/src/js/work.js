$(function tabsImg() {
  $('.work__item').on('click', function (e) {
    e.preventDefault();
    let targetEl = $(this);
    $('.work__item').each(function (el) {
      $(this).removeClass('work__focus-el');
    });
    targetEl.addClass('work__focus-el');
    $('.work__logo-item').each(function (el) {
      if (
        (targetEl.data('name') === $(this).data('name') &&
          !$(this).hasClass('hidden')) ||
        (targetEl.data('name') === 'all' && !$(this).hasClass('hidden'))
      ) {
        $(this).show();
      } else {
        $(this).hide();
      }
    });
  });
});

$(function addImg() {
  let counterBtn = 0;
  $('.work__button').on('click', function (e) {
    e.preventDefault();
    let counterEl = 0;
    $('.hidden').each(function (el) {
      let logoEl = $(this);
      if (counterEl < 12) {
        $(this).removeClass('hidden');
        if (
          $('.work__focus-el').data('name') === logoEl.data('name') ||
          $('.work__focus-el').data('name') === 'all'
        ) {
          logoEl.show();
        } else {
          logoEl.hide();
        }
        counterEl++;
      }
    });
    counterBtn++;
    if (counterBtn > 1) {
      $(this).remove();
    }
  });
});
