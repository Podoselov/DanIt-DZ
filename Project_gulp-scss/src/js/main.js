$('.header__burger').on('click', function () {
  $(this).toggleClass('header__burger--open');
  $('.header__nav-list').toggleClass('header__nav-list--open');
});
