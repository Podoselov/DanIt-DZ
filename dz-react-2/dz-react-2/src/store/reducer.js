import { combineReducers } from 'redux';
import cardsReduser from './cardsReduser.js';
import modalReduser from './modalReduser.js';

export default combineReducers({
  cards: cardsReduser,
  modal: modalReduser,
});
