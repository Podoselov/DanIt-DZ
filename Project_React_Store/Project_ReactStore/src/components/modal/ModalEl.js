import React from 'react';
import styled from 'styled-components';
import { FaTimes } from 'react-icons/fa';
import { useDispatch } from 'react-redux';
import { modalOpen } from '../../store/actions.js';

const ModalBlock = styled.div`
  position: fixed;
  z-index: 1;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgba(0, 0, 0, 0.4);
`;

const ModalContent = styled.div`
  background-color: rgb(30, 30, 32);
  top: 50%;
  transform: translateY(-50%);
  margin: auto;
  width: 35%;
  position: relative;
  color: rgb(255, 255, 255);
  h2 {
    background-color: rgb(42, 42, 46);
    padding: 26px 0 21px 29px;
    font-size: 22px;
    text-align: center;
    svg {
      padding: 0;
      margin: 0 0 0 80px;
    }
  }
  p {
    font-size: 15px;
    text-align: center;
    padding: 36px;
  }
  button {
    padding: 15px 30px;
  }
  div {
    display: flex;
    justify-content: center;
  }
`;

const ModalEl = ({
  header,
  text,
  closeButton = false,
  action,
  active,
  setActive,
}) => {
  const dispatch = useDispatch();

  return (
    <ModalBlock
      onClick={() => {
        setActive(false);
        dispatch(modalOpen(false));
      }}
      style={active ? { display: 'block' } : { display: 'none' }}
    >
      <ModalContent
        onClick={(event) => {
          event.stopPropagation();
        }}
      >
        <h2>
          {header}
          {closeButton ? (
            <FaTimes
              onClick={() => {
                setActive(false);
                dispatch(modalOpen(false));
              }}
            ></FaTimes>
          ) : null}
        </h2>
        <p>{text}</p>
        {action}
      </ModalContent>
    </ModalBlock>
  );
};

export default ModalEl;
