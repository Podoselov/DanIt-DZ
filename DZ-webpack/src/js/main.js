import jquery from './jquery-3.6.0.min.js';
import reviews from './reviews.js';
import services from './services.js';
import work from './work.js';
import '../scss/_buy.scss';
import '../scss/_color.scss';
// import '../scss/_fonts.scss';
import '../scss/_footer.scss';
import '../scss/_header.scss';
import '../scss/_news.scss';
import '../scss/_reset.scss';
import '../scss/_reviews.scss';
import '../scss/_services.scss';
import '../scss/_work.scss';
