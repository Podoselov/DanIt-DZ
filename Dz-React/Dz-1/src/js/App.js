'use strict';

import React, { Component, useState } from 'react';
import ButtonEl from '../components/button/buttonEl';
import classes from '../css/main.module.scss';
import classesButtonEl from '../components/button/buttonEl.module.scss';
import ModalEl from '../components/modal/modalEl.js';

const App = () => {
  const [button, setButtonState] = useState();

  const clickButton = (event) => {
    setButtonState(console.log(event.target));
  };

  const [modal, setModal] = useState([
    
  ]);

  const  

  return (
    <div className={classes.container}>
      <div className={classes.containerButton}>
        <ButtonEl
          classEl={classesButtonEl.button}
          backgroundColorEl='red'
          text='Open first modal'
          buttonEvent={clickButton}
        />
        <ButtonEl
          classEl={classesButtonEl.button}
          backgroundColorEl='black'
          text='Open second modal'
          buttonEvent={clickButton}
        />
      </div>
      {/* <ModalEl
        header='Do you want to delete this file?'
        text='Once you delete this file, it wonâ€™t be possible to undo this action. 
        Are you sure you want to delete it?'
      /> */}
    </div>
  );
};

export default App;
