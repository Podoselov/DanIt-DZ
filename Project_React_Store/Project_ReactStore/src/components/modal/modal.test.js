import React from 'react';
import ModalEl from './ModalEl.js';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { rootReducer } from '../../store/reducer.js';

const propsElement = {
  header: 'header',
  text: 'text',
  active: true,
  disable: false,
};

const store = createStore(rootReducer);

describe('<ModalEl component>', () => {
  it('Modal render', () => {
    const { container } = render(
      <Provider store={store}>
        <ModalEl
          active={propsElement.active}
          header={propsElement.header}
          text={propsElement.text}
        />
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should be active', () => {
    const { getByText } = render(
      <Provider store={store}>
        <ModalEl
          active={propsElement.active}
          header={propsElement.header}
          text={propsElement.text}
        />
      </Provider>
    );
    expect(getByText(/text/i)).toBeInTheDocument();
  });
});
