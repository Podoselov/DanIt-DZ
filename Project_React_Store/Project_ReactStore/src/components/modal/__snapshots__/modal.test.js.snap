// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalEl component> Modal render 1`] = `
<div
  class="sc-bdvvtL gtCsIZ"
  style="display: block;"
>
  <div
    class="sc-gsDKAQ dwbzkt"
  >
    <h2>
      header
    </h2>
    <p>
      text
    </p>
  </div>
</div>
`;
